Makefile

```diff
@@ -111,6 +112,18 @@ ifeq "$(LXD_OFFLINE)" ""
 endif
 	swagger generate spec -o doc/rest-api.yaml -w ./lxd -m
 
+.PHONY: doc
+doc:
+	@echo "Setting up documentation build environment"
+	python3 -m venv .sphinx/venv
+	. $(SPHINXENV) ; pip install --upgrade -r .sphinx/requirements.txt
+	mkdir -p .sphinx/deps/ .sphinx/themes/
+	git -C .sphinx/deps/vanilla pull || git clone https://github.com/evildmp/vanilla-sphinx-test .sphinx/deps/vanilla
+	ln -sf ../deps/vanilla/vanilla .sphinx/themes/vanilla
+	@echo "Build the documentation"
+	rm -Rf doc/html/
```

Comment:
Cleaning the output directory makes sense for a full build, but it will prevent incremental builds. This doesn't make a big difference with the current size of the documentation, but once we add pages and split up some of the long pages, build times will increase. With incremental builds, Sphinx will only build the files that changed, and this speeds up the build a lot.

So it makes sense to keep it for this target, but we should have a separate target for only building.